--- org.moreunit.swtbot.test/src/org/moreunit/refactoring/RenameClassTest.java
+++ org.moreunit.swtbot.test/src/org/moreunit/refactoring/RenameClassTest.java
@@ -62,7 +62,7 @@
	}

	@Project(
-            mainCls = "org:SomeClass",
+            mainCls = "org:OtherClass",
             testCls = "org:OtherClassTest",
             mainSrcFolder = "src",
             testSrcFolder = "test")
@@ -70,9 +70,9 @@
	public void should_rename_test_using_regex_fallback_when_cut_gets_renamed()
	{
-		renameSomeClassToAnyClassAndWaitUntilFinished();
+		renameOtherClassToAnyClassAndWaitUntilFinished();

-		assertThat(context.getCompilationUnit("org.AnyClassTest")).isNotNull();
+		assertThat(context.getCompilationUnit("org.AnyClassTest")).isNotNull();
	}

	private void renameSomeClassToAnyClassAndWaitUntilFinished()
@@ -99,5 +99,32 @@
		bot.button("Finish").click();
		SWTBotShell renameDialog = bot.activeShell();
		bot.waitUntil(Conditions.shellCloses(renameDialog), 20000);
	}
+
+	private void renameOtherClassToAnyClassAndWaitUntilFinished()
+	{
+		final SWTBotTreeItem packageItem = selectAndReturnPackageWithName("org");
+		packageItem.expand();
+		packageItem.getNode("OtherClass.java").select();
+		bot.waitUntil(new DefaultCondition()
+        {
+
+            @Override
+            public boolean test() throws Exception
+            {
+                return packageItem.getNode("OtherClass.java").isSelected();
+            }
+
+            @Override
+            public String getFailureMessage()
+            {
+                return "Node OtherClass.java not selected";
+            }
+        });
+		getShortcutStrategy().pressRenameShortcut();
+		bot.textWithLabel("New name:").setText("AnyClass");
+		bot.button("Finish").click();
+		SWTBotShell renameDialog = bot.activeShell();
+		bot.waitUntil(Conditions.shellCloses(renameDialog), 20000);
+	}

 }
